File: BackEnd/Api/Application/DataTransfer/VesselDto.cs
using Api.Domain.ValueObjects;

namespace Api.Application.DataTransfer;

public class VesselDto
{
    public required string Name { get; set; }
    public required string ImoNumber { get; set; }
    public required VesselTypeDto Type { get; set; }
    public required ShippingAgentOrganizationDto Owner { get; set; }
    public required PhysicalCharacteristics PhysicalCharacteristics { get; set; }

}

public class CreateVesselDto
{
    public required string Name { get; set; }
    public required string ImoNumber { get; set; }
    public required string Type { get; set; }
    public required string Owner { get; set; }
    public required PhysicalCharacteristics PhysicalCharacteristics { get; set; }
}

File: FrontEnd/src/model/dto/VesselDto.ts
export interface VesselDto {
    name: string;
    imoNumber: string;
    type: string;
    owner: string;
    physicalCharacteristics: {
        length: number;
        depth: number;
        draft: number;
    };
}

export interface VesselCreateDto {
    name: string;
    imoNumber: string;
    type: string;
    owner: string;
    length: number;
    depth: number;
    draft: number;
}

export interface VesselFilter {
    name?: string;
    imoNumber?: string;
    taxNumber?: string;
}